cmd/kubemark/hollow-node.go

```diff
@@ -228,14 +228,14 @@ func run(cmd *cobra.Command, config *hollowNodeConfig) error {
 			return fmt.Errorf("Failed to start fake runtime, error: %w", err)
 		}
 		defer fakeRemoteRuntime.Stop()
-		runtimeService, err := remote.NewRemoteRuntimeService(endpoint, 15*time.Second)
+		runtimeService, apiVersion, err := remote.NewRemoteRuntimeService(endpoint, 15*time.Second, "")
 		if err != nil {
 			return fmt.Errorf("Failed to init runtime service, error: %w", err)
 		}
 
 		var imageService internalapi.ImageManagerService = fakeRemoteRuntime.ImageService
 		if config.UseHostImageService {
-			imageService, err = remote.NewRemoteImageService(f.RemoteImageEndpoint, 15*time.Second)
+			imageService, err = remote.NewRemoteImageService(f.RemoteImageEndpoint, 15*time.Second, apiVersion)
```

Comment:
This may not work well for the use cases when image service is used that has lower version, for example, https://github.com/containerd/stargz-snapshotter/pull/323

I'm not sure what's the best option here - perhaps detect each version separately using the fallback.

Reply:
Okay, I decided to use the same logic to determine the API for the image service.